### DISCUSSION TITLE
Session's last line lost after a split

### DISCUSSION THREAD

**Discussion by pjfichet** [scandum/tintin#238] [Script help]
Hello !

When I do a `#split 0 2 2 26`, and close my mud session (with the appropriate mud command), the last lines sent by my mud after that command are not displayed. 

Is there anything I can do to get those lines back ?

**Comment by scandum** [comment]
It might be due to something weird your mud is doing as I can't reproduce this problem.

Do the last lines show if you use `#config convert on;quit`, and if so, what are they?

**Comment by pjfichet** [comment]
Hey ! Thank you so much for your answer :)

`\e[0m#CONFIG {CONVERT META} HAS BEEN SET TO {ON}.\e[0m` is the last line shown after `#config convert on;#send rent`.

But I've narrowed my problem: it's related to the relationship between #split, scrolling region, and sessions. Here are a few samples codes and their results:

```
#split 0 2 2 26;
#ssl {mume} {193.134.218.98} {4242};
$login
$password
rent
```
--> with the `rent` command, the session `mume` is disconnected, and tintin replaces the scrolling region within the split with the one of the #gts session. It's so quick I can't read the lines sent by the mud after I type rent (my feeling is that they're not displayed at all).

```
#ssl {mume} {193.134.218.98} {4242};
#split 0 2 2 26;
$login
$password
rent
```
-> WIth the `rent` command the session `mume` is disconnected, and tintin replaces the split and its scrolling region with the unsplit #gts session, scrolling everything up by one line. Still, the last line shown in the now gone mume session is my input `rent`. I don't even have the `#SESSION mume DIED.` line. The line `#session gts activated` is printed below the scrolled mume session.

```
#split 0 2;
#ssl {mume} {193.134.218.98} {4242};
$login
$password
rent
```
--> Tintin delegates the scroll to my terminal, and the last lines sent by the mud are correctly shown, as well as `#session mume died` and `#session gts activated`. Alas, it's not the split I want :(

I hope all this makes sense somehow.

**Comment by scandum** [comment]
I see what you mean now, I guess one work around could be:
```
#alias rent {#unsplit;#send rent}
```
There's not much I can do about it, because the vertical split code has to redraw the entire screen to work and uses the scrollback buffer to do so. While a bit overkill to see the last few lines, the following would duplicate the scrollback buffer in the startup session:
```
#event {RECEIVED LINE} {#gts #line ignore #show {%0}}
```

**Comment by pjfichet** [comment]
Hello !

Thanks for the idea `#unsplit; #send rent` is just what I need :)

Thanks again for everything !
